pywr-schema/src/nodes/river.rs

```diff
@@ -31,17 +75,82 @@ impl RiverNode {
 
 #[cfg(feature = "core")]
 impl RiverNode {
+    /// The sub-name of the output node.
+    fn loss_node_sub_name() -> Option<&'static str> {
+        Some("loss")
+    }
+
     pub fn node_indices_for_constraints(
         &self,
         network: &pywr_core::network::Network,
     ) -> Result<Vec<pywr_core::node::NodeIndex>, SchemaError> {
         let idx = network.get_node_index_by_name(self.meta.name.as_str(), None)?;
         Ok(vec![idx])
     }
+
     pub fn add_to_model(&self, network: &mut pywr_core::network::Network) -> Result<(), SchemaError> {
         network.add_link_node(self.meta.name.as_str(), None)?;
+
+        // add output node and edge
+        if self.loss.is_some() {
+            network.add_output_node(self.meta.name.as_str(), Self::loss_node_sub_name())?;
+
+            let river = network.get_node_index_by_name(self.meta.name.as_str(), None)?;
+            let loss = network.get_node_index_by_name(self.meta.name.as_str(), Self::loss_node_sub_name())?;
+            network.connect_nodes(river, loss)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn set_constraints(
+        &self,
+        network: &mut pywr_core::network::Network,
+        args: &LoadArgs,
+    ) -> Result<(), SchemaError> {
+        let river_node = network.get_node_index_by_name(self.meta.name.as_str(), None)?;
+        let current_flow_metric = MetricF64::NodeInFlow(river_node);
+
+        if let Some(loss) = &self.loss {
+            // add the flow
+            let loss_metric = match &loss.data {
```

Comment:
One issue with this approach is that the proportion is going to be a day out. The metric will always be the previous flow. That's fine, but probably needs to be made clear. 

I wonder if we need an option to use an `AggregatedNode` to do proportional losses within the LP (I.e. the same time-step). However, the interpolation is not possible then.

Reply:
Also, this would align with how the WTW node does things.